src/main/java/com/productdock/library/search/book/BookMapper.java

```diff
@@ -2,51 +2,46 @@
 
 import com.productdock.library.search.elastic.document.BookDocument;
 import com.productdock.library.search.kafka.consumer.messages.InsertBookMessage;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.List;
 
 @Component
+@AllArgsConstructor
 public class BookMapper {
 
-    @Autowired
     private BookDtoRecordMapper recordDtoMapper;
 
     public BookDto toBookDto(BookDocument bookDocument) {
         var bookDto = new BookDto();
-        bookDto.id = bookDocument.getBookId();
-        bookDto.title = bookDocument.getTitle();
-        bookDto.author = bookDocument.getAuthor();
-        bookDto.cover = bookDocument.getCover();
+        mapSimpleProperties(bookDocument, bookDto);
+        mapRecords(bookDocument, bookDto);
+        return bookDto;
+    }
 
-        var records = getBookDocumentRecords(bookDocument);
-        int availableBookCount = getAvailableBooksCount(bookDocument);
-        for (int i = 1; i <= availableBookCount; i++) {
-            var bookRecord = BookDocument.RentalState.Record.builder().status(BookStatus.AVAILABLE).build();
-            records.add(bookRecord);
-        }
+    private void mapRecords(BookDocument bookDocument, BookDto bookDto) {
+        var records = bookDocument.getRentalState().getRecords();
+        var availableBookCount = bookDocument.getRentalState().getAvailableBooksCount();
+        records.addAll(createAvailableRecords(availableBookCount));
```

Comment:
toBookDto method violates Have no side effect rule. It alters the records collection belonging to the input parameter (BookDocument). 
You can fix this by changed the way list is combined:
```
        var availableRecords = createAvailableRecords(availableBookCount);
        var allRecords = Collections.join(records, availableRecords);
```
That way bookDocument.getRentalState().getRecords() will stay unchanged , and BookDto will still get combined records.